ce/test/core/msbuild-tests.ts

```diff
@@ -26,39 +34,36 @@ describe('MSBuild Generator', () => {
 
     activation.addLocation('somepath', local.fs.file('c:/tmp'));
     activation.addPath('include', [local.fs.file('c:/tmp'), local.fs.file('c:/tmp2')]);
+    activation.addDefine('POSIX_ME_HARDER', '1');
 
-    const expected = (platform() === 'win32') ? `<?xml version="1.0" encoding="utf-8"?>
-<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <PropertyGroup Label="Locations">
-    <somepath>c:\\tmp</somepath>
-  </PropertyGroup>
-  <PropertyGroup Label="Properties">
-    <z>zse&amp;tting</z>
-    <a>ase&lt;tting</a>
-    <c>csetting</c>
-    <b>bsetting</b>
-    <prop>first;seco&gt;nd;third</prop>
-  </PropertyGroup>
-  <PropertyGroup Label="Paths">
-    <include>c:\\tmp;c:\\tmp2</include>
-  </PropertyGroup>
-</Project>` : `<?xml version="1.0" encoding="utf-8"?>
+    const fileWithNoSlash = local.fs.file('c:/tmp');
+    const fileWithSlash = local.fs.file('c:/tmp/');
+    activation.addMSBuildProperty('a', '$(a);fir$root$st', fileWithNoSlash);
+    activation.addMSBuildProperty('a', '$(a);second', fileWithNoSlash);
+    activation.addMSBuildProperty('a', '$(a);$root$hello', fileWithNoSlash);
+    activation.addMSBuildProperty('b', '$(x);first', fileWithSlash);
+    activation.addMSBuildProperty('b', '$(b);se$root$cond', fileWithSlash);
+    activation.addMSBuildProperty('a', '$(a);third', fileWithNoSlash);
+    activation.addMSBuildProperty('b', 'third', fileWithSlash);
+    activation.addMSBuildProperty('b', '$(b);$root$world', fileWithSlash);
```

Comment:
We should also add tests for xml-sensitive characters: `<`, `&`, `>`, `'`, `"`

Reply:
Given that we aren't testing the xml implementation here I don't think that's really a meaningful test but I'll add, stand by...

Reply:
I'd expect artifact to provide valid msbuild (and xml) names and values in msbuild-properties. It would be most helpful if we could check this in the artifact metadata validation when artifact is added to the registry or indexed and not every time the .props are generated.